src/utils/AddressUtils.ts

```diff
@@ -39,10 +39,14 @@ export function compareAddressesSimple(addressA?: string, addressB?: string): bo
   return addressA.toLowerCase() === addressB.toLowerCase();
 }
 
-// Converts an input (assumed to be) bytes32 string into a bytes20 string.
-// If the input is not a bytes32 but is less than type(uint160).max, then this function
-// will still succeed.
-// Throws an error if the string as an unsigned integer is greater than type(uint160).max.
+// Converts an input hex data string into a bytes32 string. Note that the output bytes will be lowercase
+// so that it naturally matches with ethers event data.
+// Throws an error if the input string is already greater than 32 bytes.
+export function toBytes32(address: string): string {
```

Comment:
Should there be a check here that the address is a valid evm address (i.e. has 20 bytes)? Or does that not matter

Reply:
We could validate w/ ethers getAddress, but that introduces an issue of what to do if it doesn't validate. Ultimately we probably don't want the bot to throw here, but we also need to return a string. I think we need to sharpen this handling up e2e but idk whether we should make the upgrade conditional on that. wdyt?

Reply:
I think a possible advantage of not checking is being able to put in either bytes20 or bytes32 addresses into here and getting back bytes32 addresses regardless.